.github/workflows/vet.yml

```diff
@@ -0,0 +1,31 @@
+name: go vet
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '**/*.go'
+
+concurrency:
+  group: ${{ github.workflow }}-$${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  vet:
+    runs-on: [ ubuntu-latest ]
+
+  steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Run go vet
+      run: ./tool/go vet ./...
+
+    - name: Build checklocks
+      run: ./tool/go build -o /tmp/checklocks gvisor.dev/gvisor/tools/checklocks/cmd/checklocks
```

Comment:
If you wanted: you can build a custom `vettool` that bundles the `checklocks` Analyzer along with the rest of the standard `go vet` analyzers, and can filter failures to be non-fatal. See `cmd/vet` in the corp repo for more details (`ignoreTests`, etc.).

Reply:
I looked at that. Is there a reason to add that cost of maintenance right now?

Reply:
Nope, feel free to ignore; just wanted to flag if you haven't seen it 👍

Reply:
Thanks yeah, I'm just not sure I want to track upstream vet changes to keep the lint set up to date. I can imagine sometime there may be a performance advantage to folding into a single pass, or if there are checks we want to disable. Right now vet passes, so that seems like a good place to maintain for now!